src/routes/StandaloneRouting.js

```diff
@@ -1,104 +1,114 @@
-
-          // Open the Request to the server for the JSON data
-          // In this case we have a server-side route called /api/
-          // which responds to GET requests with the study data
-          OHIF.log.info(`Sending Request to: ${url}`);
-          oReq.open('GET', url);
-          oReq.setRequestHeader('Accept', 'application/json')
-
-          // Fire the request to the server
-          oReq.send();
-        });
-    }
+        OHIF.log.info(JSON.stringify(oReq.responseText, null, 2));
 
-    componentDidMount() {
-        const query = qs.parse(this.props.location.search);
-        StandaloneRouting.parseQueryAndFetchStudies(query).then(studies => {
-            const updatedStudies = createDisplaySets(studies);
+        const data = JSON.parse(oReq.responseText);
+        if (data.servers && query.studyInstanceUids) {
+          const server = data.servers.dicomWeb[0];
+          server.type = 'dicomWeb';
 
-            this.setState({ studies: updatedStudies });
-        }, error => {
-            this.setState({ error });
-        });
-    }
+          const studyInstanceUids = query.studyInstanceUids.split(';');
+          const seriesInstanceUids = [];
 
-    render() {
-        if (this.state.error) {
-            return (<div>Error: {JSON.stringify(this.state.error)}</div>);
-        } else if (!this.state.studies) {
-            return (<div>Loading...</div>);
+          OHIF.studies
+            .retrieveStudiesMetadata(
+              server,
+              studyInstanceUids,
+              seriesInstanceUids
+            )
+            .then(
+              studies => {
+                resolve(studies);
+              },
+              error => {
+                reject(error);
+              }
+            );
+        } else {
+          resolve(data.studies);
         }
-
-        return (
-            <Viewer studies={this.state.studies}/>
-        );
+      });
+
+      // Open the Request to the server for the JSON data
+      // In this case we have a server-side route called /api/
+      // which responds to GET requests with the study data
+      OHIF.log.info(`Sending Request to: ${url}`);
+      oReq.open('GET', url);
+      oReq.setRequestHeader('Accept', 'application/json');
+
+      // Fire the request to the server
+      oReq.send();
+    });
+  }
+
+  componentDidMount() {
+    const query = qs.parse(this.props.location.search);
+    StandaloneRouting.parseQueryAndFetchStudies(query).then(
+      studies => {
+        const updatedStudies = createDisplaySets(studies);
+
+        this.setState({ studies: updatedStudies });
+      },
+      error => {
+        this.setState({ error });
+      }
+    );
+  }
+
+  render() {
+    if (this.state.error) {
+      return <div>Error: {JSON.stringify(this.state.error)}</div>;
+    } else if (!this.state.studies) {
+      return <div>Loading...</div>;
     }
+
+    return <Viewer studies={this.state.studies} />;
+  }
 }
 
 export default StandaloneRouting;
```

Comment:
I don't think anything changed in this file other than formatting? It took me a minute to go line by line. If I'm missing anything, please call out some line numbers 🙌

Reply:
You're right. I also rolled that back to the original version